### PR TITLE
fixes #3 - Swagger-generated client for the Skycoin REST API

### LINKED ISSUE #3: C API wrapper for the Skycoin node API
## Expected Behavior

Access to the Skycoin node REST API implemented in C code .

## Actual Behavior

Right now this is only possible using code generated by SWIG , which requires memory handles

## Specifications

  - Version: 0.25.0
  - Platform: all
  - Subsystem: all exported API sets

## Possible Solution

Generate it from Swagger specs  . Beware of CSRF .

- [ ] Use Skycoin node Swagger spec to generate initial C REST API client
- [ ] Add `make` target to automate generation in `./build`
- [ ] Aforementioned target should copy / merge relevant files into source code tree
- [ ] README explaining the process
- [ ] Mention in CHANGELOG
- [ ] Patch the result Swagger-gen REST client to support specifying node address
- [ ] Update package metadata
- [ ] Run basic Swagger tests in Travis
- [ ] Unify test code in case multiple test suites are used after generation
- [ ] Beware of CSRF auth

### PR DESCRIPTION
**Pull request by stdevMac**
Fixes #3 

Changes:
- Create a C wrapper for Skycoin Api,
- Generate swagger spec for `skycoin/skycoin` api.
- Is the First step for create clients for Skycoin Api in languages supported by swagger generator.

Does this change need to mentioned in CHANGELOG.md?
yes

Requires testing
yes

Comments about testing , should you have some
We should test correct Swagger generation and test C wrapper.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -69,6 +75,24 @@ configure-build:
...
+	@rm -rf $(SWAGGER_SPEC_DIR)/*.openapi.v2.yml
+	@rm -rf $(SWAGGER_SPEC_DIR)/*.openapi.v2.json
+	@echo "Start swagger spec generation"
+	-swagger generate spec -b github.com/skycoin/skycoin/src/api -o $(LIBSRC_ABS_PATH)/$(SWAGGER_SPEC_DIR)/skycoin.openapi.v2.yml
+	-swagger generate spec -b github.com/skycoin/skycoin/src/api -o $(LIBSRC_ABS_PATH)/$(SWAGGER_SPEC_DIR)/skycoin.openapi.v2.json
+	@echo "Finished swagger spec generation"
+
+swagger-client: ## Generate C client for accessing Skycoin REST API
+	@rm -rf $(SWAGGER_CLIENT_DIR)/cClient
+	@openapi-generator generate -g c --additional-properties=prependFormOrBodyParameters=true -o $(SWAGGER_CLIENT_DIR)/cClient -i ./lib/swagger/swagger.yml
```
Comment (olemis) [inline Makefile]: use `SWAGGER_SPEC_DIR`

File: lib/Clients/c/.openapi-generator/VERSION [hunk 2]
```diff
@@ -0,0 +1 @@
+3.3.4
```
Comment (olemis) [inline lib/Clients/c/.openapi-generator/VERSION]: Duplicate code . C client library should be under `lib/curl` folder .

**Comment by olemis** [comment]
`lib/Clients` folder should go away and its contents be moved onto `lib/curl`

File: Makefile (hunk 1 above)
Comment (stdevMac) [reply Makefile]: Is already done in commit `aac416d`

File: lib/Clients/c/.openapi-generator/VERSION (hunk 2 above)
Comment (stdevMac) [reply lib/Clients/c/.openapi-generator/VERSION]: C client moved to `lib/curl`

Comment (stdevMac) [reply lib/Clients/c/.openapi-generator/VERSION]: Really C client was regenerated by openapi-generator with current swagger status in `Skycoin Api`

**Comment by olemis** [comment]
@stdevMac conflicts

**Comment by stdevMac** [comment]
I think this conflict can be solved when PR for  issue #2031 in Skycoin is done

**Comment by olemis** [comment]
scheduling for `0.26` due to issues with Swagger generator .

**Pull request closed without merging**